File: MeetHut.Backend/ClientApp/src/app/auth/models/registration.model.ts
export class RegistrationModel {
  userName: string;
  password: string;
  email: string;
  fullName: string;

  constructor(
    userName: string,
    password: string,
    email: string,
    fullName: string
  ) {
    this.userName = userName;
    this.password = password;
    this.email = email;
    this.fullName = fullName;
  }
}

File: MeetHut.Services/Application/Models/RegistrationModel.cs
using System.ComponentModel.DataAnnotations;

namespace MeetHut.Services.Application.Models
{
    /// <summary>
    /// Registration Model
    /// </summary>
    public class RegistrationModel
    {
        /// <summary>
        /// User name
        /// </summary>
        [Required]
        [MaxLength(120)]
        public string UserName { get; set; }
        
        /// <summary>
        /// Password
        /// </summary>
        [Required]
        [MaxLength(30)]
        public string Password { get; set; }
        
        /// <summary>
        /// Email Address
        /// </summary>
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        
        /// <summary>
        /// Full name
        /// </summary>
        public string FullName { get; set; }

    }
}